### ISSUE TITLE
Merge wrong conflicts for 

### ISSUE THREAD

**Issue by boruar** [libgit2/libgit2#4241]
You are opening a _bug report_ against the libgit2 project.  If you have a
question about an API or usage, please ask on StackOverflow:
http://stackoverflow.com/questions/tagged/libgit2.  Please fill out the
reproduction steps (below) and delete this introductory paragraph.  Thanks!

### Reproduction steps
Steps to create the merge case:

1. git init test
2. cd test
3. echo line 1 >> foo.txt
4. echo line 2 >> foo.txt
5. echo line 3 >> foo.txt
6. echo line 4 >> foo.txt
7. git add foo.txt
8. git commit -m "Initial add"
9. git checkout -b task
10. echo added line on task >> foo.txt
11. git commit -a -m "changed on source"
12. git mv foo.txt bar.txt
13. git commit -m "renamed on source"
14. git checkout master
15. echo added line on master >> foo.txt
16. git commit -a -m "changed on source"

Merge from:  "refs/heads/task"

Merge  call:
int git_merge(
	git_repository *repo,
	const git_annotated_commit **their_heads,
	size_t their_heads_len,
	const git_merge_options *merge_opts,
	const git_checkout_options *given_checkout_opts)

Used options:
git_merge_options merge_opts= GIT_MERGE_OPTIONS_INIT;
merge_opts.flags = GIT_MERGE_FIND_RENAMES;
merge_opts.rename_threshold= 50;
merge_opts.target_limit = 200;

git_checkout_options given_checkout_opts = GIT_CHECKOUT_OPTIONS_INIT;
given_checkout_opts.checkout_strategy = GIT_CHECKOUT_SAFE;

### Expected behavior
* git status --short 
D foo.txt
UU bar.txt

*  Ancestor contributor content for bar.txt:
line 1
line 2
line 3
line 4
*  Theirs contributor content for bar.txt
line 1
line 2
line 3
line 4
added line on task
*  Ours contributor content for bar.txt
line 1
line 2
line 3
line 4
added line on master

### Actual behavior
* git status --short
UA bar.txt
UD foo.txt 

The problem is that the index doesn't have the expected info. 

When the modified content on task and modified content on master can be combined (the file merge ca be resolved automatically) the merge result is OK. So the rename is properly detected.

### Version of libgit2 (release number or SHA1)
sha: 2fcb870

### Operating system(s) tested
Windows 10

**Comment by ethomson** [comment]
Thanks for the helpful repro case.

libgit2 includes content size limits on rename detection to prevent false positives.  These are generally considered to be reasonable, since a very small file should probably not be considered for rename detection:  consider a file that is one byte.  Any change to that could be a rename.  Git previously did this as well, but it looks like these have been removed or relaxed.

If you repeat your test with a larger file (say 25 lines or so) the do you see libgit2 behaving the way you expect?

**Comment by boruar** [comment]
Yes, it's also reproduced with 25 lines. It's reproduced with any number of lines. 

The problem occurs always that the modified content on task and the modified content on master cannot be combined automatically. When the moved files must appear as UU conflict after the merger
